Bearing and Distance algorithm rewrite / cleanup

StewLG opened the pull request:
The home bearing and distance algorithms in PlayUAV have been under a lot of suspicion for potentially being defective; there are dozens of bug reports in this area, and little progress in understanding. These rewrites may not correct the original bug(s), but they are arguably cleaner and (in the case of distance), slightly more accurate, and seem worth moving to for those reasons alone.

StewLG committed:
The home bearing and distance algorithms in PlayUAV have been under a lot of suspicion for potentially being defective; there are dozens of bug reports in this area, and little progress in understanding. These rewrites may not correct the original bug(s), but they are arguably cleaner and (in the case of distance), slightly more accurate, and seem worth moving to for those reasons alone.

TobiasBales commented on src/math3d.c:
```diff
@@ -604,6 +604,19 @@ void PLANE3D_Init(PLANE3D_PTR plane, POINT3D_PTR p0, VECTOR3D_PTR normal,
+{
+    // convert ang to radians
+    float theta = (angle) * M_PI / 180.0;
+    return theta;
```
This could just be `return (angle) * M_PI / 180.0;`

StewLG replied:
Sure.

TobiasBales commented on src/math3d.c:
```diff
@@ -622,6 +635,7 @@ void Build_Sin_Cos_Tables(void) {
 
 } // end Build_Sin_Cos_Tables
 
+
```
no need for the extra new line, but don't care too much either ways

StewLG replied:
These are just accidents, not based on any strong opinion. Changed.

TobiasBales commented on src/osdproc.c:
```diff
@@ -692,79 +708,155 @@ void draw_time() {
+                                               float lon_one, 
+                                               float lat_two, 
+                                               float lon_two)
+{
```
currently we have the opening `{` on the line of the function definition (which I prefer) but we could talk about general code standards like this if you care.

I started using uncrustify but it does some odd things every now and then.

I used clang_format quite a bit lately and it seems to work nicely.
It formats stuff a bit weird (for my taste at least), but it is consistent and fast.

My suggestion would be to ignore style things like this, decide on a tool to enforce styling, choose a style and just reformat all our code at once

TobiasBales replied:
Maybe rename the parameters to `lat_from, lon_from, lat_to, lon_to`?
It does not matter which way around (I think) but it makes it easier
when reading the function signature because then it is suggested,
that one param goes in one place and the other in the other.

StewLG replied:
I'm more used to the brace style I used, but I always try to stick to the prevailing standard. My deviation here is just habit. Fixed.

As for the parameters, I changed them - it really doesn't matter much. I was just thinking that since direction ("from" & "to") aren't important, don't name the parameters that way. In other functions like bearing, direction **is** important, and I wanted to make a distinction.

Again, no big deal.

TobiasBales commented on src/osdproc.c:
```diff
@@ -692,79 +708,155 @@ void draw_time() {
+// Why bundle these things together? And what does CWH stand for anyhow? -- SLG
+void draw_CWH(void) {
+  
+  set_home_position_if_unset();
```
I think we should do this in the loop before starting to draw panels on the screen.
That way you can use the info in any panel and the initialization of state could be moved to a central point (over time)

StewLG replied:
I lacked the courage to do this initially, thanks for the nudge. Done.

TobiasBales reviewed (changes requested):
I really really like the changes, thanks a ton!

StewLG committed:
Changes requested by Tobias

Mostly just syntax, but the change to drop draw_CWH entirely, and move all its functions to the draw loop, is a notable improvement I was hesitant to make until he prodded me.

TobiasBales commented:
Really really nice work!
